src/lib.rs

```diff
@@ -304,16 +303,20 @@ fn dt_to_generalized(dt :&DateTime<Utc>) -> GeneralizedTime {
 }
 
 impl Certificate {
-	/// Generates a new self-signed certificate from the given parameters
-	pub fn from_params(params :CertificateParams) -> Self {
-		let (key_pair, key_pair_serialized) = DynKeyPair::generate(&params.alg);
+	/// Generates a new certificate from the given parameters and key pair
+	pub fn from_params_and_keypair(params :CertificateParams, key_pair :KeyPair) -> Self {
```

Comment:
Instead of this function, could you add a `key_pair :Option<KeyPair>` field to `CertificateParams` instead?

Reply:
Sure thing. Like this https://github.com/est31/rcgen/commit/33ea827ce1dcbee50d88dea2adfd8b4c548dbc2c#diff-b4aea3e418ccdb71239b96952d9cddb6L307?

Reply:
Yup, that's good.